frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -0,0 +1,191 @@
+
+import SearchBarModal from "../Modal/SearchBarModal";
+import SearchBarModalContent from "./SearchBarModalContent";
+import CustomButton from "./CustomButton";
+import "./Searchbar.modules.scss";
+
+const presentationList = [
+  "react presentation",
+  "JS presentation",
+  "SASS presentation",
+  "Design presentation",
+];
+
+function Searchbar() {
+  const [showFilter, setShowFilter] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [searchValue, setSearchValue] = useState("");
+  const [type, setType] = useState("presentation");
+  const [category, setCategory] = useState("back-end");
+  const [searchList, setSearchList] = useState<string[]>([]);
+
+  function handleChange(e: React.FormEvent<HTMLInputElement>) {
+    setShowFilter(false);
+    setSearchValue(e.currentTarget.value);
+    setShowModal(true);
+    setSearchList(
+      presentationList.filter((item) =>
+        item.toLocaleLowerCase().includes(searchValue.toLowerCase())
+      )
+    );
+  }
+
+  function toggleFilter() {
+    if (!showFilter) {
+      setShowModal(true);
+    } else {
+      setShowModal(false);
+    }
+    setShowFilter((state) => !state);
+    setSearchList([]);
+    setSearchValue("");
+  }
+
+  function selectTypeHandler(e: React.MouseEvent<HTMLButtonElement>) {
+    setType(e.currentTarget.value);
+  }
+
+  function selectCategoryHandler(e: React.MouseEvent<HTMLButtonElement>) {
+    setCategory(e.currentTarget.value);
+  }
+
+  function modalHandler() {
+    setShowFilter(false);
+    setSearchList([]);
+    setSearchValue("");
+    setShowModal(false);
+  }
+
+  function clickCaptureHandler() {
+    setShowModal(true);
+    setShowFilter(false);
+    setSearchList([]);
+    setSearchValue("");
+  }
+
+  return (
+    <div className="searchbar">
+      <div className="searchbar__form">
+        <input
+          type="search"
+          name="search"
+          className="searchbar__form--input"
+          placeholder="Search"
+          autoComplete='off'
+          value={searchValue}
+          onChange={handleChange}
+          onClickCapture={clickCaptureHandler}
+        />
+        {searchList.length > 0 && searchValue && (
+          <div className="searchbar__form--searchList">
+            {searchList.slice(0, 5).map((item, index) => (
+              <div className="searchbar__form--searchItem" key={index}>
+                {item}
+              </div>
+            ))}
+          </div>
+        )}
+        <div className="searchbar__container">
+          <button
+            className="searchbar__container--btn filter"
+            onClick={toggleFilter}
+          >
+            <FilterIcon />
+          </button>
+          <button className="searchbar__container--btn search">
+            <SearchIcon />
+          </button>
+        </div>
+        <div
+          className={`searchbar__filter--container ${showFilter && "active"} `}
+        >
+          <div className="searchbar__filterType">
```

Comment:
suggestion: you can move the filtering and categories part into a separate component and pass a prop for change handling, to clean this component up a bit.